track_info.c

```diff
@@ -52,16 +48,18 @@ struct track_info *track_info_new(const char *filename)
 	uint64_t uid = atomic_fetch_add_explicit(&cur_uid, 1, memory_order_relaxed);
 	BUG_ON(uid == 0);
 
-	struct track_info *ti;
-	ti = xnew(struct track_info, 1);
-	ti->uid = uid;
+	struct track_info_priv *priv = xnew(struct track_info_priv, 1);
+	atomic_store_explicit(&priv->ref_count, 1, memory_order_relaxed);
```

Comment:
>As I understand it, we can't have any races here, so a simple assignment is good enough.

You're misunderstanding. A simple assignment has stronger synchronization properties than the code above. Compare:

```c
#include <stdatomic.h>

struct X {
	_Atomic int j;
};

void f(struct X *x)
{
	x->j = 0;
}

void g(struct X *x)
{
	atomic_store_explicit(&x->j, 0, memory_order_relaxed);
}
```

```asm
f:
	xorl	%eax, %eax
	xchgl	%eax, (%rdi)
	retq
g:
	movl	$0, (%rdi)
	retq
```